src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.MaxResponseHeadersLength.cs

```diff
@@ -100,8 +97,34 @@ public void ValidValue_SetGet_Roundtrips(int validValue)
                     });
                 }
             });
-
         }
 
+        public static IEnumerable<object[]> ResponseWithManyHeadersData
+        {
+            get
+            {
+                // Generate a response with lots of headers so that the
+                // total is greater than 1024 bytes but less than 2048 bytes.
+                var buffer = new StringBuilder();
+                buffer.Append("HTTP/1.1 200 OK\r\n");
+                for (int i = 0; i < 50; i++)
+                {
+                    buffer.Append($"Custom-{i}: 1234567890\r\n");
+                }
+                buffer.Append("Content-Length: 0\r\n\r\n");
+                string _responseWithManyHeaders = buffer.ToString();
+                Assert.InRange(_responseWithManyHeaders.Length, 1025, 2048);
+
+                // Failure case: response headers must be <= 1024 bytes.
+                {
+                    yield return new object[] { _responseWithManyHeaders, 1, false };
+                }
+
+                // Success case: response headers must be <= 2048 bytes.
+                {
+                    yield return new object[] { _responseWithManyHeaders, 2, true };
+                }
```

Comment:
I revised the tests.  I now have 3 tests to test "less than", "equal to", "greater than" the max setting.